core/src/mindustry/ui/dialogs/ModsDialog.java

```diff
@@ -133,11 +144,118 @@ void setup(){
                             }, t2 -> Core.app.post(() -> modError(t2)));
                         });
                     }).margin(12f);
-                });
 
+                    t.row();
+
+                    t.button("$mod.featured.title", Icon.star, bstyle, () -> {
+                        try{
+                            dialog.hide();
+                            BaseDialog dialog2 = new BaseDialog("$mod.featured.dialog.title");
+                            dialog2.cont.table(table -> {
+                                table.left();
+                                table.image(Icon.zoom);
+                                searchBar = table.field(searchtxt, res -> {
+                                    searchtxt = res;
+                                    rebuildBrowser[0].run();
+                                }).growX().get();
+                            }).fillX().padBottom(4);
+
+                            dialog2.cont.row();
+
+                            dialog2.cont.pane(tablebrow -> {
+                                tablebrow.margin(10f).top();
+                                rebuildBrowser[0] = () -> {
+                                    tablebrow.clear();
+                                    String searchString = searchtxt.toLowerCase();
+                                    Core.net.httpGet("https://raw.githubusercontent.com/Anuken/MindustryMods/master/mods.json", response -> {
+                                        if(response.getStatus() != HttpStatus.OK){
+                                            ui.showErrorMessage(Core.bundle.format("connectfail", response.getStatus()));
+                                        }else{
+                                            Json json = new Json();
+                                            Seq<ModListing> listings = json.fromJson(Seq.class, ModListing.class, response.getResultAsString());
+                                            if(ModListing mod : listings){
```

Comment:
This, uh, doesn't compile.

Reply:
What? OhnoOhnoOhnoOhno. What does intellij say?

Reply:
Its very strange since it compiles in my case :/

Reply:
Whats does intellij say?

Reply:
There's no way `if(ModListing mod : listings){` compiles, that statement doesn't make any sense. You're confusing an if with a `for`.

Reply:
ohno, somehow i managed to change the fors with ifs, but in the version on my PC isnt, very strange